labman/db/tests/test_study.py

```diff
@@ -82,6 +83,107 @@ def test_samples(self):
         exp_samples = ['1.SKB1.640202', '1.SKD1.640179', '1.SKM1.640183']
         self.assertEqual(s.samples('1.64'), exp_samples)
 
+    def test_samples_with_specimen_id(self):
+        s = Study(1)
+
+        # HACK: the Study object in labman can't modify specimen_id_column
+        # hence we do this directly in SQL, if a test fails the transaction
+        # will rollback, otherwise we reset the column to NULL.
+        sql = """UPDATE qiita.study
+                 SET specimen_id_column = %s
+                 WHERE study_id = 1"""
+        with sql_connection.TRN as TRN:
+            TRN.add(sql, ['anonymized_name'])
+
+            exp_samples = ['SKB1', 'SKB2', 'SKB3',
+                           'SKB4', 'SKB5', 'SKB6',
+                           'SKB7', 'SKB8', 'SKB9',
+                           'SKD1', 'SKD2', 'SKD3',
+                           'SKD4', 'SKD5', 'SKD6',
+                           'SKD7', 'SKD8', 'SKD9',
+                           'SKM1', 'SKM2', 'SKM3',
+                           'SKM4', 'SKM5', 'SKM6',
+                           'SKM7', 'SKM8', 'SKM9']
+            self.assertEqual(s.samples(), exp_samples)
+            exp_samples = ['SKB1', 'SKB2', 'SKB3',
+                           'SKB4', 'SKB5', 'SKB6',
+                           'SKB7', 'SKB8', 'SKB9']
+            self.assertEqual(s.samples(limit=9), exp_samples)
+            exp_samples = ['SKB1', 'SKB2', 'SKB3',
+                           'SKB4', 'SKB5', 'SKB6',
+                           'SKB7', 'SKB8', 'SKB9']
+            self.assertEqual(s.samples('SKB'), exp_samples)
+            exp_samples = ['SKB1', 'SKB2', 'SKB3']
+            self.assertEqual(s.samples('SKB', limit=3), exp_samples)
+            exp_samples = ['SKM1', 'SKM2', 'SKM3',
+                           'SKM4', 'SKM5', 'SKM6',
+                           'SKM7', 'SKM8', 'SKM9']
+            self.assertEqual(s.samples('SKM'), exp_samples)
+            exp_samples = ['SKM1', 'SKM2', 'SKM3',
+                           'SKM4', 'SKM5', 'SKM6',
+                           'SKM7', 'SKM8', 'SKM9']
+            # case insensitive
+            self.assertEqual(s.samples('Skm'), exp_samples)
+            self.assertEqual(s.samples('64'), [])
+
```

Comment:
Is there a test of a `limit` but no `term`?

Reply:
Yes there's a test in line 111.

Reply:
oh, sorry!

Reply:
👍